### PR TITLE
Remove nb_classes

### PR DESCRIPTION
**Pull request by carlini**
Remove nb_classes argument from CW and JSMA attacks. Infer it from the shape of the result of the model.

### CODE REVIEW

File: cleverhans/attacks.py [hunk 1]
```diff
@@ -172,6 +172,27 @@ def generate_np(self, x_val, **kwargs):
 
         return self.sess.run(x_adv, feed_dict)
 
+    def get_labels(self, x, kwargs):
```
Comment (goodfeli) [inline cleverhans/attacks.py]: docstring please

Comment (goodfeli) [inline cleverhans/attacks.py]: maybe rename to `get_or_guess_labels` since it can use the predictions instead of the labels?

**Comment by goodfeli** [comment]
Looks good apart from the two comments I made

**Comment by carlini** [comment]
(For github: This fixes #217.)

**Pull request merged**